src/gui/updatenotifier.cpp

```diff
@@ -29,37 +29,47 @@ void UpdateNotifier::checkForUpdates(const AccountPtr &account)
+                    connect(updateJob, &JsonJob::finishedSignal, this, [updateJob, this] {
+                        if (updateJob->httpStatusCode() == 200) {
+                            const auto data = updateJob->data().value("channels"_L1).toObject().value(channel()).toObject();
+                            if (!data.isEmpty()) {
+                                const auto currentVersion = data.value("current_version"_L1).toString();
+                                const auto url = QUrl(data.value("url"_L1).toString());
+                                if (version() < QVersionNumber::fromString(currentVersion)) {
+                                    auto notification = SystemNotificationRequest{tr("Update available"),
+                                        tr("A new version %1 is available. You are using version %2.",
+                                            "The first placeholder is the new version, the second one the current version")
+                                            .arg(currentVersion, version().toString()),
+                                        Resources::FontIcon(u'')};
 
-                            const QString buttonText = tr("Open Download Page");
-                            notification.setButtons({buttonText});
-                            auto *sysNotification = ocApp()->systemNotificationManager()->notify(std::move(notification));
-                            connect(sysNotification, &SystemNotification::buttonClicked, this, [buttonText, url](const QString &button) {
-                                if (button == buttonText) {
-                                    QDesktopServices::openUrl(url);
+                                    const QString buttonText = tr("Open Download Page");
+                                    notification.setButtons({buttonText});
+                                    auto *sysNotification = ocApp()->systemNotificationManager()->notify(std::move(notification));
+                                    connect(sysNotification, &SystemNotification::buttonClicked, this, [buttonText, url](const QString &button) {
+                                        if (button == buttonText) {
+                                            QDesktopServices::openUrl(url);
+                                        }
+                                    });
                                 }
-                            });
+                            }
+                        } else {
+                            qCWarning(lcUpdateNotifier) << u"Update check failed with HTTP code" << updateJob->httpStatusCode();
                         }
-                    }
+                    });
+                    updateJob->start();
                 } else {
-                    qCWarning(lcUpdateNotifier) << u"Update check failed with HTTP code" << job->httpStatusCode();
+                    qCWarning(lcUpdateNotifier) << u"Update check config.json with HTTP code" << job->httpStatusCode();
```

Comment:
are the warnings user facing? because then they'd be very technical 🙈

Reply:
only logs